src/ocean/net/util/QueryParams.d

```diff
@@ -445,3 +445,77 @@ unittest
         }
     }
 }
+
+
+unittest
+{
+    scope params = new FullQueryParamSet(',', ':', "hello", "world");
+    params.parse("hello:1,world:2,Die:Katze,tritt");
+    test!("==")(params["hello"], "1");
+    test!("==")(params["world"], "2");
+    test(!("Die" in params));
+    test(!("tritt" in params));
+    test!("==")(params.remaining_elements.length, 2);
+
+    with (params.remaining_elements[0])
+    {
+        test!("==")(key, "Die");
+        test!("==")(val, "Katze");
+    }
+
+    with (params.remaining_elements[1])
+    {
+        test!("==")(key, "tritt");
+        test!("is")(val, val.init);
+    }
+
+    bool[4] found;
+
+    foreach (key, val; params)
```

Comment:
(Probably unrelated to this PR.) Interesting API, that `opIn_r` doesn't report these extra values as being present, but `opApply` iterates over them.

Reply:
Good point. IMHO they should not show up via `in` or `foreach`. It seems too risky to change it just for correctness, though, it could break some mission-critical server code. We can fix this if someone actually needs the correct behaviour.

Reply:
:+1: 
Maybe something for ocean v5.x.x.

Reply:
Issue for v5.0.0: https://github.com/sociomantic-tsunami/ocean/issues/559.